app/models/embedded_ansible_worker.rb

```diff
@@ -0,0 +1,27 @@
+class EmbeddedAnsibleWorker < MiqWorker
+  require_nested :Runner
+
+  self.required_roles = ['embedded_ansible']
+
+  def start_runner
+    self.class::Runner.start_worker(worker_options)
+    # TODO: return supervisord pid
+  end
+
+  def stop
+    super
+    # stop supervisord service
+  end
+
+  def kill
+    # Does the base class's kill -9 work on the supervisord process as we want?
```

Comment:
This and setting the right worker pid is not handled in this PR.  For now, the server's pid is stored in the worker row but we have disabled monitoring of this pid's memory.